# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.h
# Commit: 2f1e8f5b2fc2
# Full Hash: 2f1e8f5b2fc2f5a28fd2f25019108e56de0fbb44
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Rework GMP PluginID plumbing in VideoCondit. r=bwc
#   
#   This prepares VideoConduit for the decoder/encoder factory and its handling of
#   PluginID.
#   
# ==============================================================================

diff -r 777632b3319a -r 2f1e8f5b2fc2 dom/media/webrtc/libwebrtcglue/VideoConduit.h
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Fri Feb 05 10:51:01 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Mon Feb 08 17:23:44 2021 +0100
@@ -211,7 +211,7 @@
 
   void OnSinkWantsChanged(const rtc::VideoSinkWants& wants);
 
-  uint64_t CodecPluginID() override;
+  bool HasCodecPluginID(uint64_t aPluginID) override;
 
   void SetPCHandle(const std::string& aPCHandle) override {
     MOZ_ASSERT(NS_IsMainThread());
@@ -465,9 +465,9 @@
   std::unique_ptr<webrtc::VideoEncoder> mEncoder;  // only one encoder for now
   std::vector<std::unique_ptr<webrtc::VideoDecoder>> mDecoders;
   // Main thread only
-  uint64_t mSendCodecPluginID = 0;
+  nsTArray<uint64_t> mSendCodecPluginIDs;
   // Main thread only
-  uint64_t mRecvCodecPluginID = 0;
+  nsTArray<uint64_t> mRecvCodecPluginIDs;
 
   // Main thread only
   std::string mPCHandle;
